bevy_state doesn't compile when only the `bevy_app` feature is enabled

## Bevy version

ecdd1624f302c5f71aaed95b0984cbbecf8880b7

## What you did

Create an empty rust project. Edit `Cargo.toml` to add a dependency on `bevy_state` with `default_features = false` and `features = [ "bevy_app" ]`

```rust
bevy_state = { git = "https://github.com/bevyengine/bevy.git", default-features = false, features = [
	"bevy_app",
], }
```

Try to compile

## What went wrong

Crate won't compile with the error:

```
error[E0046]: not all trait items implemented, missing: `enable_state_scoped_entities`
```

## Additional information

This is because `AppExtStates` has a method `enable_state_scoped_entities`, but it's implementation on `App` is behind the `bevy_hierarchy` feature flag.

If this is required, I think either `bevy_app` should automatically enable the `bevy_hierarchy` feature, or the method in the trait should be feature gated as well.

--- a/crates/bevy_state/src/app.rs
+++ b/crates/bevy_state/src/app.rs
@@ -177,7 +177,6 @@ impl AppExtStates for App {
         self
     }
 
-    #[cfg(feature = "bevy_hierarchy")]
     fn enable_state_scoped_entities<S: States>(&mut self) -> &mut Self {
         self.main_mut().enable_state_scoped_entities::<S>();
         self

